### PR TITLE
Manila: added ListDetail for detailed Share listing 

### PR DESCRIPTION
**Pull request by gman0**
For #1094

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

- example response: https://github.com/openstack/manila/blob/master/api-ref/source/samples/shares-list-detailed-response.json
- `get_all` function: https://github.com/openstack/manila/blob/master/manila/share/api.py#L1469

### CODE REVIEW

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 1]
```diff
@@ -66,6 +69,81 @@ func Create(client *gophercloud.ServiceClient, opts CreateOptsBuilder) (r Create
 	return
 }
 
+// ListOpts holds options for listing Shares. It is passed to the
+// shares.List function.
+type ListOpts struct {
```
Comment (jtopjian) [inline openstack/sharedfilesystems/v2/shares/requests.go]: As best as I can tell, this looks correct. I've found some fields also defined in:

* https://github.com/openstack/manila/blob/e4acaa5bf14c1e188a21aab5f15d9613ebb0aa6e/manila/api/v1/shares.py#L123
* https://github.com/openstack/manila/blob/e4acaa5bf14c1e188a21aab5f15d9613ebb0aa6e/manila/api/v1/shares.py#L187

The v1 code is called from v2 by way of inheritance: https://github.com/openstack/manila/blob/e4acaa5bf14c1e188a21aab5f15d9613ebb0aa6e/manila/api/v2/shares.py#L421-L450

File: openstack/sharedfilesystems/v2/shares/results.go [hunk 2]
```diff
@@ -101,6 +104,84 @@ type CreateResult struct {
...
+	}
+
+	q := currentURL.Query()
+	q.Set("offset", mark)
+	currentURL.RawQuery = q.Encode()
+	return currentURL.String(), nil
+}
+
+// LastMarker returns the last offset in a ListResult.
+func (r SharePage) LastMarker() (string, error) {
```
Comment (jtopjian) [inline openstack/sharedfilesystems/v2/shares/results.go]: I realize this was copied from `sharenetworks` but I wanted to flag it for a sanity check. 

My interpretation of this is that Gophercloud will request an `offset` of `9223372036854775807` in order to deliberately make an erroneous call which will return no shares. Is this really the only way to determine this? Wouldn't this cause a lot of unnecessary GETs over time?

File: openstack/sharedfilesystems/v2/shares/requests.go (hunk 1 above)
Comment (gman0) [reply openstack/sharedfilesystems/v2/shares/requests.go]: I've added `metadata`, `extra_specs` and the missing fields from your second link. Thanks!

File: openstack/sharedfilesystems/v2/shares/results.go (hunk 2 above)
Comment (gman0) [reply openstack/sharedfilesystems/v2/shares/results.go]: Should be fixed now: `LastMarker` now returns an "invalid marker" (-1) which is recognized by `NextPageURL` which then returns an empty string - this stops the page iteration. Thanks!

**Comment by gman0** [comment]
@jtopjian PTAL

File: openstack/sharedfilesystems/v2/shares/requests.go (hunk 1 above)
Comment (jtopjian) [reply openstack/sharedfilesystems/v2/shares/requests.go]: Nice - thank you.

File: openstack/sharedfilesystems/v2/shares/results.go (hunk 2 above)
Comment (jtopjian) [reply openstack/sharedfilesystems/v2/shares/results.go]: Thank you for this as well!

**Comment by gman0** [comment]
Thanks for the review and merge!

**Pull request merged**